pkg/expr/nodes.go

```diff
@@ -500,8 +525,90 @@ func WideToMany(frame *data.Frame) ([]mathexp.Series, error) {
 		if err != nil {
 			return nil, err
 		}
+		if fixSeries != nil {
+			fixSeries(s, frame.Fields[tsSchema.ValueIndices[0]])
+		}
 		series = append(series, s)
 	}
 
 	return series, nil
 }
+
+type seriesFixFunction = func(series mathexp.Series, valueField *data.Field)
+
+func checkIfNeedsToBeFixed(frames []*data.Frame) seriesFixFunction {
+	// determine whether all potential series have 0 labels. If there is at least one value field with non-zero labels, the slice will be empty
+	valueFieldsWithEmptyLabels := make([]*data.Field, 0, len(frames)*len(frames[0].TimeSeriesSchema().ValueIndices)) // make a best guess for the size
+LABELS:
+	for _, frame := range frames {
```

Comment:
The more common case (most datasources) is that it won't need to be "fixed". Therefore might it be better to use the existing loop in WideToMany to see if it needs be done, and then if it needs to be done, then call the fixing pass?